frontend/packages/console-app/src/hooks/useNotificationPoller.ts

```diff
@@ -0,0 +1,123 @@
+import * as React from 'react';
+import * as _ from 'lodash';
+import { useTranslation } from 'react-i18next';
+// eslint-disable-next-line @typescript-eslint/ban-ts-comment
+// @ts-ignore: FIXME out-of-sync @types/react-redux version as new types cause many build errors
+import { useDispatch } from 'react-redux';
+import { Alert } from '@console/dynamic-plugin-sdk';
+import { PrometheusRulesResponse } from '@console/dynamic-plugin-sdk/src/lib-core';
+import {
+  alertingErrored,
+  alertingLoaded,
+  alertingLoading,
+  setAlertCount,
+} from '@console/internal/actions/observe';
+import { coFetchJSON } from '@console/internal/co-fetch';
+import { PrometheusEndpoint } from '@console/internal/components/graphs/helpers';
+import {
+  getAlertsAndRules,
+  silenceMatcherEqualitySymbol,
+} from '@console/internal/components/monitoring/utils';
+import {
+  getAlertName,
+  getAlertTime,
+} from '@console/shared/src/components/dashboard/status-card/alert-utils';
+import { useNotificationAlerts } from '@console/shared/src/hooks/useNotificationAlerts';
+
+/** Fetches notification alerts from redux store and updates the notification count.
+ * Polls the Prometheus and Alertmanager for notification alerts AND silences, stores the
+ * results into redux, adjusts the API polling frequency based on the poll success.
+  @returns Nothing, its a side-effect-driven hook.
```

Comment:
nit: the `@returns` should also be prepended with a `*`, see [a tsdoc example](https://tsdoc.org/pages/tags/returns/)